src/index.ts

```diff
@@ -26,16 +18,18 @@ class VueBreadcrumbs implements PluginObject<ComponentOptions<Vue>> {
             .flatMap((route: RouteRecord) => {
               let routeRecord: RouteRecord[] = [];
 
-              if (route.meta && route.meta.breadcrumb && route.meta.breadcrumb.parent) {
+              if (route.meta?.breadcrumb?.parent) {
                 const matched = this.$router.resolve({name: route.meta.breadcrumb.parent}).route.matched
 
-                routeRecord = [...matched, ...routeRecord];
+                routeRecord = [...matched];
               }
 
-              routeRecord.push(route)
-              return routeRecord
+              return [...routeRecord, route]
             })
-            .map(resolvePath);
+            .map((route: RouteRecord) => route.path.length === 0
```

Comment:
It seems to me that this construction is redundant for the sake of changing one value in the object; here we can leave the construction from the function `resolvePath` ?

Reply:
This construction is the same `resolvePath` function.
It seemed to me that it is redundant to put in separate function.
As you wish

Reply:
It is normal practice to change the property immutably. 
Performance won't suffer (:

Reply:
well, let it be so :)